### PR TITLE
Wakatime top languages support

### PR DESCRIPTION
**Pull request by dron22**
This PR adds support for top languages to chroma, to match behavior of pygments.

- Adds missing file extension support to existing lexers.
- Adds missing lexers if not existing in chroma.
- Does not include tests, as this will be tested in `wakatime-cli` repo. PR will be opened, once this PR is merged. 

Changes are reflected in updated `lexers/TODO.md` document.

### CODE REVIEW

**Comment by gandarez** [comment]
Are tests not working for this PR?

File: lexers/c/crontab.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package c
+
+import (
+	. "github.com/alecthomas/chroma" // nolint
+	"github.com/alecthomas/chroma/lexers/internal"
+)
+
+// Crontab lexer.
+var Crontab = internal.Register(MustNewLexer(
```
Comment (gandarez) [inline lexers/c/crontab.go:9]: Is this a new lexer?

File: lexers/g/go.go [hunk 2]
```diff
@@ -13,7 +13,7 @@ var Go = internal.Register(MustNewLexer(
 	&Config{
 		Name:      "Go",
 		Aliases:   []string{"go", "golang"},
-		Filenames: []string{"*.go"},
+		Filenames: []string{"*.go", "go.mod"},
```
Comment (gandarez) [inline lexers/g/go.go]: Didn't find it on pygments nor Chroma.

File: lexers/g/gosu.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+		Filenames: []string{"*.gs", "*.gsx", "*.gsp", "*.vark"},
+		MimeTypes: []string{"text/x-gosu"},
+	},
+	Rules{
+		"root": {},
+	},
+))
+
+// GosuTemplate lexer.
+var GosuTemplate = internal.Register(MustNewLexer(
```
Comment (gandarez) [inline lexers/g/gosu.go]: What do you think to move it to a new file called `gosutemplate.go`?

File: lexers/l/lasso.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+// Lasso lexer.
+var Lasso = internal.Register(MustNewLexer(
+	&Config{
+		Name:    "Lasso",
+		Aliases: []string{"lasso", "lassoscript"},
+		Filenames: []string{
+			"*.lasso",
+			"*.lasso[89]",
+			// alias filesnames from pygments
```
Comment (gandarez) [inline lexers/l/lasso.go]: There's a dedicated attribute to inform aliases for filenames -> `AliasFilenames`

File: lexers/m/marko.go [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package m
+
+import (
+	. "github.com/alecthomas/chroma" // nolint
+	"github.com/alecthomas/chroma/lexers/internal"
+)
+
+// Marko lexer.
+var Marko = internal.Register(MustNewLexer(
```
Comment (gandarez) [inline lexers/m/marko.go:9]: Didn't find it on pygments nor Chroma.

File: lexers/m/mustache.go [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package m
+
+import (
+	. "github.com/alecthomas/chroma" // nolint
+	"github.com/alecthomas/chroma/lexers/internal"
+)
+
+// Mustache lexer.
+var Mustache = internal.Register(MustNewLexer(
```
Comment (gandarez) [inline lexers/m/mustache.go:9]: Didn't find it on pygments nor Chroma.

File: lexers/p/pawn.go [hunk 7]
```diff
@@ -27,3 +27,16 @@ var Pawn = internal.Register(MustNewLexer(
 
 	return 0
 }))
+
+// SourcePawn lexer.
+var SourcePawn = internal.Register(MustNewLexer(
```
Comment (gandarez) [inline lexers/p/pawn.go]: What do you think to move it to a new file called `sourcepawn.go`?

File: lexers/s/sketchdrawing.go [hunk 8]
```diff
@@ -0,0 +1,18 @@
+package s
+
+import (
+	. "github.com/alecthomas/chroma" // nolint
+	"github.com/alecthomas/chroma/lexers/internal"
+)
+
+// SketchDrawing lexer.
+var SketchDrawing = internal.Register(MustNewLexer(
```
Comment (gandarez) [inline lexers/s/sketchdrawing.go:9]: Didn't find it on pygments nor Chroma.

File: lexers/s/sublimetextconfig.go [hunk 9]
```diff
@@ -0,0 +1,18 @@
+package s
+
+import (
+	. "github.com/alecthomas/chroma" // nolint
+	"github.com/alecthomas/chroma/lexers/internal"
+)
+
+// SublimeTextConfig lexer.
+var SublimeTextConfig = internal.Register(MustNewLexer(
```
Comment (gandarez) [inline lexers/s/sublimetextconfig.go:9]: Didn't find it on pygments nor Chroma.

File: lexers/s/svelte.go [hunk 10]
```diff
@@ -0,0 +1,18 @@
+package s
+
+import (
+	. "github.com/alecthomas/chroma" // nolint
+	"github.com/alecthomas/chroma/lexers/internal"
+)
+
+// Svelte lexer.
+var Svelte = internal.Register(MustNewLexer(
```
Comment (gandarez) [inline lexers/s/svelte.go:9]: Didn't find it on pygments nor Chroma.

File: lexers/t/twig.go [hunk 11]
```diff
@@ -10,7 +10,7 @@ var Twig = internal.Register(MustNewLexer(
 	&Config{
 		Name:      "Twig",
 		Aliases:   []string{"twig"},
-		Filenames: []string{},
+		Filenames: []string{"*.twig"},
```
Comment (gandarez) [inline lexers/t/twig.go:13]: Didn't find it in Pygments.

File: lexers/x/xaml.go [hunk 12]
```diff
@@ -0,0 +1,19 @@
+package x
+
+import (
+	. "github.com/alecthomas/chroma" // nolint
+	"github.com/alecthomas/chroma/lexers/internal"
+)
+
+// XAML lexer.
+var XAML = internal.Register(MustNewLexer(
```
Comment (gandarez) [inline lexers/x/xaml.go:9]: Didn't find it on pygments nor Chroma.

File: lexers/x/xslt.go [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+	. "github.com/alecthomas/chroma" // nolint
+	"github.com/alecthomas/chroma/lexers/internal"
+)
+
+// XSLT lexer.
+var XSLT = internal.Register(MustNewLexer(
+	&Config{
+		Name:      "XSLT",
+		Aliases:   []string{"xslt"},
+		Filenames: []string{"*.xsl", "*.xslt", "*.xpl"},
```
Comment (gandarez) [inline lexers/x/xslt.go:14]: missing comment here `xpl is XProc`

**Review by gandarez — COMMENTED** [review]
Left many comments ❓

File: lexers/g/go.go (hunk 2 above)
Comment (dron22) [reply lexers/g/go.go]: This is wakatime specific :wink:

Comment (dron22) [reply lexers/g/go.go]: https://github.com/wakatime/wakatime/blob/2e636d389bf5da4e998e05d5285a96ce2c181e3d/wakatime/stats.py#L173

File: lexers/m/marko.go (hunk 5 above)
Comment (dron22) [reply lexers/m/marko.go:9]: This is a wakatime top language: https://wakatime.com/leaders/languages.

File: lexers/m/mustache.go (hunk 6 above)
Comment (dron22) [reply lexers/m/mustache.go:9]: This is a wakatime top language: https://wakatime.com/leaders/languages.

File: lexers/s/sketchdrawing.go (hunk 8 above)
Comment (dron22) [reply lexers/s/sketchdrawing.go:9]: This is a wakatime top language: https://wakatime.com/leaders/languages.

File: lexers/s/sublimetextconfig.go (hunk 9 above)
Comment (dron22) [reply lexers/s/sublimetextconfig.go:9]: This is a wakatime top language: https://wakatime.com/leaders/languages.

File: lexers/s/svelte.go (hunk 10 above)
Comment (dron22) [reply lexers/s/svelte.go:9]: This is a wakatime top language: https://wakatime.com/leaders/languages.

File: lexers/t/twig.go (hunk 11 above)
Comment (dron22) [reply lexers/t/twig.go:13]: This is a wakatime top language: https://wakatime.com/leaders/languages.

File: lexers/x/xaml.go (hunk 12 above)
Comment (dron22) [reply lexers/x/xaml.go:9]: This is a wakatime top language: https://wakatime.com/leaders/languages.

File: lexers/c/crontab.go (hunk 1 above)
Comment (dron22) [reply lexers/c/crontab.go:9]: This is a wakatime top language: https://wakatime.com/leaders/languages.

File: lexers/g/go.go (hunk 2 above)
Comment (gandarez) [reply lexers/g/go.go]: I'm not too sure we have to change Chroma's implementation adding WakaTime specific needs to it. If it's really necessary and it's missing on Pygments let's open a PR to them and discuss this specific one before merging it.

Comment (dron22) [reply lexers/g/go.go]: After your comment, I thought about it. If we consider, that pygments and chroma are used for language detection as well as language parsing and highlighting, it becomes obvious, that `go.mod` does not fit into golang lexer, as it does not contain golang code. Will move it into wakatime-cli code. :+1:

**Comment by dron22** [comment]
@gandarez Move `go.mod` support to wakatime-cli (https://github.com/wakatime/wakatime-cli/pull/158)

**Comment by gandarez** [comment]
Tests are not running for this PR.. so I forced it by running:
```bash
curl -u <token>: -d "build_parameters[CIRCLE_JOB]=build" https://circleci.com/api/v1.1/project/github/wakatime/chroma/tree/wakatime-top-languages-support
```

**Review by gandarez — APPROVED** [review]

**Comment by dron22** [comment]
> Tests are not running for this PR.. so I forced it by running:
> 
> ```shell
> curl -u <token>: -d "build_parameters[CIRCLE_JOB]=build" https://circleci.com/api/v1.1/project/github/wakatime/chroma/tree/wakatime-top-languages-support
> ```

ok, thank you! and good to know this little trick :+1:

**Pull request merged**